mtp/mtp_db.cpp

```diff
@@ -1,57 +1,79 @@
 // Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
+#include <algorithm>
 #include "mtp_db.hpp"
 
 namespace mtp
 {
+    namespace filename_to_handle
+    {
+        inline Handle &getHandle(std::map<std::filesystem::path, Handle>::iterator iter)
+        {
+            return iter->second;
+        }
+    } // namespace filename_to_handle
+
+    namespace handle_to_filename
+    {
+        inline auto &getIter(std::map<Handle, std::map<std::filesystem::path, Handle>::iterator>::iterator iter)
+        {
+            return iter->second;
+        }
+        inline const Handle &getHandle(
+            std::map<Handle, std::map<std::filesystem::path, Handle>::iterator>::const_iterator iter)
+        {
+            return iter->first;
+        }
+        inline const std::filesystem::path &getFilename(
+            std::map<Handle, std::map<std::filesystem::path, Handle>::iterator>::const_iterator iter)
+        {
+            return iter->second->first;
+        }
+
+    } // namespace handle_to_filename
+
     std::optional<std::filesystem::path> FileDatabase::get_filename(Handle handle) const
     {
-        if (const auto result = entries.find(DatabaseEntry{handle}); result != entries.end()) {
-            return result->get_filename();
+        const auto handleToFilenameIter = handleToFilename.find(handle);
+        if (handleToFilenameIter != handleToFilename.end()) {
+            return handle_to_filename::getFilename(handleToFilenameIter);
         }
         return std::nullopt;
     }
     bool FileDatabase::remove(const Handle handle)
     {
-        return entries.erase(DatabaseEntry{handle}) != 0;
+        const auto handleToFilenameIter = handleToFilename.find(handle);
+        if (handleToFilenameIter != handleToFilename.end()) {
```

Comment:
[minor]
If you negated the condition and returned false, a smaller portion of code would be indented - better readability

Reply:
Done